uast/transformer/positioner/positions.go

```diff
@@ -70,19 +93,77 @@ func fillLineColFromOffset(idx *positionIndex, pos *uast.Position) error {
 	return nil
 }
 
+func fromUnicodeOffset(idx *positionIndex, pos *uast.Position) error {
+	off, err := idx.RuneOffset(int(pos.Offset))
+	if err != nil {
+		return err
+	}
+	pos.Offset = uint32(off)
+	return fromOffset(idx, pos)
+}
+
+// runeSpan represents a sequence of UTF8 characters of the same size in bytes.
+type runeSpan struct {
+	runeOff  int // index of the first rune
+	byteOff  int // bytes offset of the first rune
+	runes    int // number of runes
+	runeSize int // in bytes
+}
+
 type positionIndex struct {
 	offsetByLine []int
+	runeSpans    []runeSpan
```

Comment:
Could probably be simply `spans`, since the rune-ness is a property of the value not the field.